human_general_array_analysis/2_identify_generic_genes_pathways.py

```diff
@@ -0,0 +1,732 @@
+)
+
+template_DE_stats = pd.read_csv(
+    template_DE_stats_filename, sep="\t", header=0, index_col=0
+)
+
+selected = template_DE_stats[
+    (template_DE_stats["adj.P.Val"] < 0.05) & (abs(template_DE_stats["logFC"]) > 1)
+]
+print(selected.shape)
+
+# + magic_args="-i grp_metadata_filename -i project_id -i base_dir -i local_dir -i num_runs" language="R"
+#
+# source(paste0(base_dir, '/generic_expression_patterns_modules/DE_analysis.R'))
+#
+# # Files created: "<local_dir>/DE_stats/DE_stats_simulated_data_SRP012656_<n>.txt"
+# for (i in 0:(num_runs-1)){
+#     simulated_data_filename <- paste(local_dir,
+#                                      "pseudo_experiment/selected_simulated_data_",
+#                                      project_id,
+#                                      "_",
+#                                      i,
+#                                      ".txt",
+#                                      sep = "")
+#
+#     get_DE_stats_limma(grp_metadata_filename,
+#                        project_id,
+#                        simulated_data_filename,
+#                        "simulated",
+#                        local_dir,
+#                        i)
+# }
+# -
+
+# ### Rank genes
+
+analysis_type = "DE"
+template_DE_stats, simulated_DE_summary_stats = ranking.process_and_rank_genes_pathways(
+    template_DE_stats_filename,
+    local_dir,
+    num_runs,
+    project_id,
+    analysis_type,
+    col_to_rank_genes,
+    logFC_name,
+    pvalue_name,
+)
+
+# ### Gene summary table
+
+# +
+summary_gene_ranks = ranking.generate_summary_table(
+    template_DE_stats_filename,
+    template_DE_stats,
+    simulated_DE_summary_stats,
+    col_to_rank_genes,
+    local_dir,
+    "gene",
+    params,
+)
+
+summary_gene_ranks.head()
+# -
+
+# Check if there is an NaN values, there should not be
+summary_gene_ranks.isna().any()
+
+# Create `gene_summary_fielname`
+summary_gene_ranks.to_csv(gene_summary_filename, sep="\t")
+
+# ### Compare gene ranking
+# Studies have found that some genes are more likely to be differentially expressed even across a wide range of experimental designs. These *generic genes* are not necessarily specific to the biological process being studied but instead represent a more systematic change.
+#
+# We want to compare the ability to detect these generic genes using our method vs those found by [Crow et. al. publication](https://www.pnas.org/content/pnas/116/13/6491.full.pdf). Their genes are ranked 0 = not commonly DE; 1 = commonly DE. Genes were ranked by the number differentially expressed gene sets a gene appeared in across 600 experiments.
+
+# +
+# Get generic genes identified by Crow et. al.
+DE_prior_filename = params["reference_gene_filename"]
+ref_gene_col = params["reference_gene_name_col"]
+ref_rank_col = params["reference_rank_col"]
+
+figure_filename = f"gene_ranking_{col_to_rank_genes}.svg"
+
+corr, shared_ranking = ranking.compare_gene_ranking(
```

Comment:
This plot actually looks more well-behaved to me than most of the RNA-seq plots you've made in that most values are fairly close to the diagonal. This one is using array data to train SOPHIE, right?

Reply:
Yes, here we trained SOPHIE on array data from Crow et al and then compared the common findings to those found by manually curating experiments from Crow et al.